app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -258,6 +260,12 @@ class NavigationOverlayFragment : Fragment() {
         }
     }
 
+    private fun observePinnedTileRemoval(): Disposable {
+        return pinnedTileChannel.removeTileEvents.subscribe { tileToRemove ->
+            pinnedTileChannel.removeContent(tileToRemove)
+        }
```

Comment:
Hm, so we listen to the adapter, react to that information in the fragment, then use it to call back into the adapter.  I think we can simplify this.  What if we remove `DefaultChannel#removeContent`, then call straight into the repo from here?

Reply:
Makes sense to me